capstone/src/main/java/com/google/step/clients/ScoreClient.java

```diff
@@ -0,0 +1,160 @@
+package com.google.step.clients;
+
+import com.google.step.data.RestaurantPageViews;
+import com.google.step.data.RestaurantScore;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Collections;
+
+
+/**
+ * Will take care of calculating the score for each restaurant.
+ */
+public class ScoreClient {
+
+    private final float minimum = 10.0f;
+    private final MetricsClient metricsClient = new MetricsClient();
+
+    /**
+     * Only method to be called and calculates the scores of each restaurant.
+     * @return an unmodifiable list of restaurantScore class.
+     */
+    public List<RestaurantScore> calculateScores() {
+        Map<Long, Double> scoreMap = new HashMap<>();
+        List<Long> restaurantIds = new ArrayList<>();
+
+        List<RestaurantPageViews> allPageViews = metricsClient.getAllPageViews("restaurantKey");
+
+        double systemAverage = getSystemAverage(allPageViews);
+
+        // Calculate score
+        for (RestaurantPageViews restaurant: allPageViews) {
+            double restaurantAverage = calculateRestaurantAveragePageViews(restaurant);
+
+            // Since the data is sorted we can just get the latest pageView for that restaurant.
+            int lastPageViewIndex = restaurant.getPageViews().size() - 1;
+            int latestPageView = restaurant.getPageViews().get(lastPageViewIndex).getCount();
+
+            double score = calculateRawScore(latestPageView, restaurantAverage, systemAverage);
+            restaurantIds.add(Long.parseLong(restaurant.getName()));
+            scoreMap.put(Long.parseLong(restaurant.getName()), score);
+        }
+
+        standardizeScores(scoreMap, restaurantIds);
+        normalizeScores(scoreMap, restaurantIds);
+
+        // Build the scores return list.
+        List<RestaurantScore> scores = new ArrayList<>();
+        for (Long restaurantId: restaurantIds) {
+            scores.add(new RestaurantScore(restaurantId, scoreMap.get(restaurantId)));
+        }
+
+        return Collections.unmodifiableList(scores);
+    }
+
+    /**
+     * Gets the average pageViews for the entire system.
+     * @param allPageViews list of restaurantPageViews holding all the pageViews.
+     * @return double representing the average pageViews for the system.
+     */
+    private double getSystemAverage(List<RestaurantPageViews> allPageViews) {
+        double sum = 0;
+        double count = 0;
+
+        for (RestaurantPageViews restaurant : allPageViews) {
+            for (int i = 0; i < restaurant.getPageViews().size(); i++) {
+                sum += restaurant.getPageViews().get(i).getCount();
+                count++;
+            }
+        }
+
+        return sum / count;
```

Comment:
Can count be zero?

Reply:
No its not possible for the count to be zero since in that case the restaurant instance wouldn't even exist. The minimum for count would be 1 in this case.

Reply:
What if the `List<RestaurantPageViews>` input is an empty list?

Reply:
Fixed in the only public method.